modules/cert-manager-acme-dns01-aws.adoc

```diff
@@ -0,0 +1,187 @@
++
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1alpha1
+kind: CertManager
+metadata:
+  name: cluster
+  ...
+spec:
+  ...
+  controllerConfig:                                <1>
+    overrideArgs:
+      - '--dns01-recursive-nameservers=1.1.1.1:53' <2>
+      - '--dns01-recursive-nameservers-only'       <3>
+----
+<1> Add the `spec.controllerConfig` section.
+<2> Provide a comma-separated list of `<host>:<port>` nameservers to query for the DNS-01 self check.
+<3> Specify to only use recursive nameservers instead of checking the authoritative nameservers associated with that domain.
++
+// TODO: Upstream docs say that this might take longer using this - determine if that's true/state here if so in a future update
+
+.. Save the file to apply the changes.
+
+. Create a namespace for the issuer.
+
+.. Create a YAML file that defines a `Namespace` object:
++
+.Example `namespace.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: my-issuer-namespace <1>
+----
+<1> Specify the namespace for the issuer.
+
+.. Create the `Namespace` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f namespace.yaml
+----
+
+. Create a secret to store your AWS credentials in.
+
+.. Create a YAML file that defines a `Secret` object:
++
+.Example `secret.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "aws-secret"                          <1>
+  namespace: my-issuer-namespace              <2>
+type: Opaque
+data:
+  awsSecretAccessKey: <aws_secret_access_key> <3>
+----
+<1> Specify the name for the secret.
+<2> Specify the namespace for the secret.
+<3> Replace `<aws_secret_access_key>` with your AWS secret access key.
+
+.. Create the `Secret` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f secret.yaml
+----
+
+. Create an issuer.
+
+.. Create a YAML file that defines the `Issuer` object:
++
+.Example `issuer.yaml` file
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging                                        <1>
+  namespace: my-issuer-namespace                                   <2>
+spec:
+  acme:
+    server: https://acme-staging-v02.api.letsencrypt.org/directory <3>
+    email: "<email_address>"                                       <4>
+    privateKeySecretRef:
+      name: <secret_private_key>                                   <5>
+    solvers:
+    - dns01:
+        route53:
+          accessKeyID: <aws_key_id>                                <6>
+          hostedZoneID: <hosted_zone_id>                           <7>
+          region: us-east-1
+          secretAccessKeySecretRef:
+            name: "aws-secret"                                     <8>
+            key: "awsSecretAccessKey"                              <9>
+      selector:
+        dnsNames:
+        - '<domain_name>'                                          <10>
```

Comment:
@bergerhoffer you can remove the "selector" stanza, it is not needed for our document.
"dnsNames" under "selector" require an exact match (see https://cert-manager.io/docs/configuration/acme/#dns-names says "dnsNames take an exact match") and the usage of "selector" field actually occupies significant documenting: after "The solvers stanza has an optional selector field" in https://cert-manager.io/docs/configuration/acme/ , https://cert-manager.io/docs/configuration/acme/ uses half page for "selector" field and a standalone section for "dnsNames". The function of "dnsNames" in "Issuer" yaml is not the same function of "dnsNames" field in Certificate. If you want to understand more, you can look at my https://docs.google.com/document/d/1ZUzGBIZVOwqQrEB-M0zE9aAjK8_ue7Gfsxv_jm4XX7Q/ dns01 examples.